File: ios/Models/ProcessorHttpCallInfo.swift
import Foundation

public class ProcessorHttpCallInfo: JSONParams {
  private var object: [AnyHashable: Any?]

  override init(target: [AnyHashable: Any]?) {
    self.object = [:]
    
    super.init(target: target)
    
    self.parseJSON()
  }

  private func parseJSON() {
    let epochSecond = self.getParam("epochSecond", type: Int.self)
    let tid = self.getParam("tid", type: String.self)
    let path = self.getParam("path", type: String.self)
    let requestMethod = self.getParam("requestMethod", type: String.self)
    let date = self.getParam("date", type: String.self)

    self.object["epochSecond"] = self.defaultIfNull(epochSecond, defaultValue: -1)
    self.object["tid"] = tid
    self.object["path"] = path
    self.object["requestMethod"] = requestMethod
    self.object["date"] = date
  }

  func getDict() -> NSDictionary {
    return NSDictionary(dictionary: self.object as [AnyHashable: Any])
  }
}

File: android/src/main/java/com/azifacemobile/models/ProcessorHttpCallInfo.java
package com.azifacemobile.models;

import com.azifacemobile.models.abstracts.CommonParams;
import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.WritableMap;

import org.json.JSONObject;

public class ProcessorHttpCallInfo extends CommonParams {
  private final WritableMap object;

  public ProcessorHttpCallInfo(JSONObject target) {
    super(target);

    this.object = Arguments.createMap();

    this.parseJSON();
  }

  private void parseJSON() {
    Integer epochSecond = this.getParam("epochSecond", Integer.class);
    String tid = this.getParam("tid", String.class);
    String path = this.getParam("path", String.class);
    String requestMethod = this.getParam("requestMethod", String.class);
    String date = this.getParam("date", String.class);

    this.object.putInt("epochSecond", this.defaultIfNull(epochSecond, -1));
    this.object.putString("tid", tid);
    this.object.putString("path", path);
    this.object.putString("requestMethod", requestMethod);
    this.object.putString("date", date);
  }

  public WritableMap getMap() {
    return this.object.copy();
  }
}
